test/color/test_hls.py

```diff
@@ -173,3 +173,18 @@ def op_script(data: torch.Tensor) -> torch.Tensor:
             actual = op_script(data)
             expected = kornia.hls_to_rgb(data)
             assert_allclose(actual, expected)
+
+    def test_nan(self):
+        np.random.seed(0)
+
+        img = (np.random.randint(0, 256, [1000, 1000, 3], np.uint8) / 255.0).astype(np.float32)
+
+        cv_img = cv2.cvtColor(img, cv2.COLOR_RGB2HLS)
```

Comment:
I think, it is better to generate image offline, convert with OpenCV, save result and compare to it, using load?
Moreover, I prefer much smaller image.

Otherwise we are dependent on cv2 just for a single test.

Reply:
I followed the test provided by @Dipet in PR #306 to be able to reproduce the NaN issue.The other tests also make use of random numbers and up to my opinion can generate the same problem. He provided the random seed so we can reproduce it in this specific case. Shall I try to generate an example with 5x5 matrix?

Also shall I avoid the random generation in the rest of tests?

Reply:
The problem is not random generation, but use of OpenCV in the core of the library.
To reproduce @Dipet case is enough to find crashing case and store it somewhere, I think.

Reply:
Try this values
```
img = np.array(
        [
            [
                [0.85490197, 0.29411766, 0.84705883,],
                [0.4627451, 0.2784314, 0.15686275,],
                [0.29803923, 0.29803923, 0.6,],
                [0.07058824, 0.09019608, 0.58431375,],
                [0.3529412, 0.69411767, 0.23137255,],
            ],
            [
                [0.7058824, 0.6, 0.15686275,],
                [0.3254902, 0.44705883, 0.21176471,],
                [0.9372549, 0.2509804, 0.19607843,],
                [0.6509804, 0.6509804, 0.6509804,],
                [0.4509804, 0.14901961, 0.62352943,],
            ],
        ],
        dtype=np.float32,
    )
```

Reply:
I updated the test for nan with values that always generate nan values. Still compared against OpenCV for keeping the style of the other tests. I will change the rest of the tests in the following days

Reply:
Pushed in the last commit hardcoded